libraries/core_libs/consensus/src/pbft/period_data_queue.cpp

```diff
@@ -38,7 +38,7 @@ bool PeriodDataQueue::push(PeriodData &&period_data, const dev::p2p::NodeID &nod
   const auto period = period_data.pbft_blk->getPeriod();
   std::unique_lock lock(queue_access_);
 
-  if (period != std::max(period_, max_pbft_size) + 1) {
+  if (period != std::max(period_, max_pbft_size) + 1 && (queue_.empty() && period != max_pbft_size + 2)) {
```

Comment:
Maybe there is a better way to fix it. So I often got an error when we are just about to get sync. We are pushing a block to the queue at the moment where previous block was retrieved from the queue but not yet pushed in the chain. It is a valid scenario that the block we are pushing could be either 1 or two blocks ahead if one block is just about to get pushed in the chain.
There is no danger in this change because in PbftManager::processPeriodData we do another check that the proper block is in the queue. 
I added a comment about this.